server/lib/src/constants/acp.rs

```diff
@@ -2064,3 +2067,91 @@ lazy_static! {
         ..Default::default()
     };
 }
+
+// Application Create/Manage
+//   needs to be able to assign to entry managed by
+lazy_static! {
+    pub static ref IDM_ACP_APPLICATION_CREATE_DL6: BuiltinAcp = BuiltinAcp {
+        classes: vec![
```

Comment:
Previous comment was "how do you envisage these access controls working with human roles and tasks", and I think you need to answer that because these access controls otherwise seem confused.

Reply:
Ok, I can think about two possibilities:

Use case 1, the IDM administrator creates the application, links the group, and sets up the members.
Use case 2, there is a person managing service X, e.g. mail. The IDM administrator creates the app, links a group, 
and delegates group permission for the group to the X service manager who sets up the members.

Reply:
Okay, so I actually think both possibilities are valid. We have the ability to create new applications (which we will give to an idm_application_admins group, and by default, idm_admins is a member of that, see https://github.com/kanidm/kanidm/blob/master/server/lib/src/constants/groups.rs#L203 and https://github.com/kanidm/kanidm/blob/master/server/lib/src/constants/acp.rs#L594 ) and then for group 2, we already have a delegation feature thanks to our "Entry Managed By" feature. You can see an example of this https://github.com/kanidm/kanidm/blob/master/server/lib/src/constants/acp.rs#L1047 

So we'll need to support  Attribute::EntryManagedBy on the LDAP application class, as well as make EntryManagedBy writable by the application_admins group, then there is a second ACP for ApplicationEntryManager which has the receiver as EntryManager and similar or constrained application rights. Does that seem reasonable?

Reply:
Ok, it is clearer now but I still have questions:

For case 1:
I have merged `IDM_ACP_APPLICATION_{CREATE,MANAGE,DELETE}_DL7` into `IDM_ACP_APPLICATION_MANAGE_DL7`. The receiver is the `idm_application_admins` builtin group and has "full control" over the application entry.

For case 2:
I don't see the need for `Attribute::EntryManagedBy` in the application entry nor the need for a second ACP having `EntryManager` as receiver. IIUC a `idm_application_admins_group` member will create the application and will link a group, and then will delegate the *linked group* management, not the application entry. And as you said we already have `IDM_ACP_GROUP_ENTRY_MANAGER_V1` to allow the delegated person to modify the linked group members. Unless you think the delegated person should be able to modify the linked group in the application entry.

Reply:
But linked group is about the users who can *access* the application, not the group that *owns and manages* the application integration itself. So that's why I think there needs to be a separation.

Reply:
Ok, understood. I have added it.